lib/tasks/eu_exit_business_finder.rake

```diff
@@ -0,0 +1,108 @@
+namespace :eu_exit_business_finder do
+  task update_subscriber_list_titles_and_facet_values: [:environment] do
+    subscriber_lists = SubscriberList
+      .where("links::text != '{}'")
+      .where("ARRAY(SELECT json_object_keys(links)) <@ Array[:keys]", keys: [:facet_values])
+    update_facet_values(subscriber_lists)
+  end
+
+  # Iterate through the affected subscriber lists and if we encounter any
+  # facet values that need replacing with a newer one, we insert the new one
+  # to the array (inserted 'in place' to retain original order). Then remove
+  # the old facet values in bulk and write only the unique values that remain.
+  def update_facet_values(subscriber_lists)
+    subscriber_lists.each do |list|
+      links = list.links[:facet_values][:any]
+      updated_links = links.map { |original_value| replace_facet_value(original_value) || original_value }
+      updated_links = (updated_links - EuExitFacetMigrationConfig::facet_values_to_remove).uniq
+      list.write_attribute('links', facet_values: { any: updated_links })
+      list.write_attribute('title', title_from_facets(updated_links))
+      list.save
+    end
+  end
+
+  def replace_facet_value(original_facet_value)
+    EuExitFacetMigrationConfig::facet_values_to_replace
+      .select { |replace_value| replace_value[:old_facet_value] == original_facet_value }
+      .map { |replace_value| replace_value[:new_facet_value] }[0]
```

Comment:
Is there an easier way to do this?

Would this work:

```
EuExitFacetMigrationConfig::facet_values_to_replace.map |facet_value_to_replace| do
   if facet_value_to_replace[:old_facet_value] != original_facet_value
       return facet_value_to_replace[:new_facet_value]
   end
   original_facet_value
end
```

Reply:
If you make `facet_values_to_replace` into a hash (rather than an array of hashes) with the key being the `old_facet_value` and the value being the `new_facet_value` then this method won't be needed.